godot/ModularWeapons/ModularWeapon/src/Projectile/Laser/LaserTracer.gd

```diff
@@ -17,28 +17,28 @@ func trace_path(lifetime_actual: float) -> Array:
 
 	var current_time := 0.0
 
-	var collided := false
+	var has_collided := false
 	
 	while current_time < lifetime:
 		current_time += TIME_STEP
 
 		var planned_movement := _update_movement(TIME_STEP)
 		
-		if not collided and current_time < lifetime_actual:
+		if not has_collided and current_time < lifetime_actual:
 			var collision := move_and_collide(planned_movement)
 		
 			if not collision:
 				positions.append(global_position)
 			else:
-				positions.append(collision.position)
-				emit_signal("collided", collision.collider, collision.position)
+				positions.append(collision.get_position())
+				emit_signal("collided", collision.get_collider(), collision.get_position())
```

Comment:
Directly using the signal: `collided.emit(collision.get_collider(), collision.get_position())`